### DISCUSSION TITLE
Dropped frames

### DISCUSSION THREAD

**Discussion by ChaeYoungKim-snu** [miniscope/minian#187] [Q&A]
Hi,

I am working with the miniscope recording in sync with electrophysiological data. 
Recently I got to know that there are natural drops of frames in every several thousands of frames, mainly caused by USB power. So I started to detect dropped frames, and I found 39 dropped frames from 11455 miniscope frames. ( so the correct total should be 11455 +39). I already preprocessed all image data using Minian without noticing this problem. 

At this point, I am thinking how to fix my data. 
method 1) Insert "interpolated" frames on the missing frame sites, and re-run Minian analysis
method 2) Do not re-run Minian, but insert "interpolated" value on the calcium trace
method 3) Do not re-run Minian, but insert "nan" value on the missing frame sites and run analysis

I wonder how sensitive the Minian processing is, in case of dropped frames? There are mostly 1 frame each drop and sometimes max 3 frames (~100ms) in rare cases. Do you think i should rerun Minian with interpolated frames (method 1)? Or   method 2 or 3 is also acceptable, given the calcium dynamics?

To give you a general idea of frame drops of my data, here I attach the total number of frames (x axis) and dropped frames (red lines). There is a slight mismatch of each frame number because x-axis is from miniscope (with all dropped frames) and red lines (from another hardware without drops), but please ignore this difference. 

Thank you for your help

Best regards,
Chae

![MD15_drops](https://user-images.githubusercontent.com/47932984/109800853-03d3b000-7c1e-11eb-9fc9-c742e3b799fb.png)

**Comment by phildong** [comment]
I think it heavily depends on how sensitive your downstream analysis depends on minian output.
Personally I think minian (cnmf algorithm) is pretty insensitive to single-frame drops, and I bet if you insert interploated frames you would get results that's indistinguishable from what you already have.
The only thing I would caution against is inserting nan/zero frames. Most algorithms in minian won't be happy with Nan values, and might be subject to artifacts if things drop to zero/black for a frame.
